press/scripts/migrate.py

```diff
@@ -473,30 +524,60 @@ def executed_from_temp_dir():
 	return cur_file.startswith(temp_dir)
 
 
-if __name__ in ("__main__", "frappe.integrations.frappe_providers.frappecloud"):
+@click.command()
+def main():
+	global has_external_files, external_db_path, external_public_files_path, external_private_files_path, external_config_file_path
+	local_site = ""
 	if executed_from_temp_dir():
 		current_file = os.path.abspath(__file__)
 		atexit.register(cleanup, current_file)
 
-	try:
-		local_site = sys.argv[1]
-	except Exception:
-		local_site = input("Name of the site you want to migrate: ").strip()
+	frappe_provider = click.prompt(
+		"Frappe provider (default: frappecloud.com)", default="frappecloud.com"
+	)
 
-	try:
-		frappe_provider = sys.argv[2]
-	except Exception:
-		frappe_provider = "frappecloud.com"
+	restore_choice = click.prompt(
+		"Do you want to restore from external files? (yes/no)", default="no"
+	)
+	if restore_choice.lower() in ["yes", "y"]:
+		has_external_files = True
+		try:
+			external_db_path = click.prompt("Enter full path to the external database file")
+			external_file_checker(external_db_path, "database")
```

Comment:
You are using exception handling as a control flow, and as a means to pass validation message. A better way to do it would be to just pass values, you'd even remove a level of indentation that way.